File: 프로그래머스/0/120866. 안전지대/안전지대.java
class Solution {
    public int solution(int[][] board) {
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[i][j] == 1) continue;

                for (int k = i - 1; k < i + 2; k++) {
                    for (int l = j - 1; l < j + 2; l++) {
                        if (k >= 0 && k < board.length && l >= 0 && l < board.length && board[k][l] == 1) {
                            board[i][j]--;
                        }
                    }

                }
            }

        }
        int answer = 0;
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[i][j] == 0) answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv0/120866. 안전지대/안전지대.py
def solution(board):
    # 상 우상 우 우하 하 좌하 좌 좌상
    dx = [-1, -1, 0, 1, 1, 1, 0, -1]
    dy = [0, 1, 1, 1, 0, -1, -1, -1]
    n = len(board)
    
    check_board = [[True] * n for _ in range(n)]
    
    for i in range(n):
        for j in range(n):
            # 지뢰의 위치인 경우
            if board[i][j]:
                check_board[i][j] = False
                # 탐색
                for d in range(8):
                    nx = i + dx[d]
                    ny = j + dy[d]
                    if 0 <= nx < n and 0 <= ny < n and check_board[nx][ny]:
                        check_board[nx][ny] = False
                
    return sum(map(sum, check_board))